onebusaway-android/src/main/java/com/joulespersecond/seattlebusbot/PreferencesActivity.java

```diff
@@ -149,7 +162,10 @@ public boolean onPreferenceChange(Preference preference, Object newValue) {
             }
         } else if (preference.equals(analyticsPref) && newValue instanceof Boolean) {
             Boolean isAnalyticsActive = (Boolean) newValue;
```

Comment:
The SharedPreference value for analyticsPref doesn't persist until `onPreferenceChange()` returns (see [Android docs](http://developer.android.com/reference/android/preference/Preference.OnPreferenceChangeListener.html), so you should be ok doing a synchronous call to ObaAnalytics.reportEventWithCategory() without having to do `forceReportEventWithCategory()`.  See also my comment on `forceReportEventWithCategory()`.  Please test to double-check the Android docs are correct ;).